opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/MLOperator.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.planner.physical;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.ml.common.dataframe.DataFrame;
+import org.opensearch.ml.common.dataframe.Row;
+import org.opensearch.ml.common.output.MLOutput;
+import org.opensearch.ml.common.output.MLPredictionOutput;
+import org.opensearch.sql.ast.expression.Literal;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.planner.physical.PhysicalPlan;
+import org.opensearch.sql.planner.physical.PhysicalPlanNodeVisitor;
+
+/**
+ * ml-commons Physical operator to call machine learning interface to get results for
+ * algorithm execution.
+ */
+@RequiredArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class MLOperator extends MLCommonsOperatorActions {
```

Comment:
what is difference with MLCommonsOperator, deptecated MLCommonsOperator?

Reply:
MLOperator is totally new operator for all ml algorithms from ml-commons. I don't want to mix it with old operator as there is a big gap between them, and also for smoothly deprecating old operator in the future.